nominal/core/_utils.py

```diff
@@ -0,0 +1,47 @@
+from typing import Iterable, Protocol, TypeVar, runtime_checkable
+
+from .._utils import logger
+
+T = TypeVar("T")
+
+
+@runtime_checkable
+class HasRid(Protocol):
+    rid: str
```

Comment:
Previously I was just using a large union, but then I have to import all the resource types. This is a simple way to tag which types can have their RIDs extracted without actual hierarchies.